### PR TITLE
KAA-1232 Client authentication key pair generation build paramether

### PR DESCRIPTION
**Pull request by Dubland**
/cc @Slash32 @RostakaGmfun

### CODE REVIEW

File: client/client-multi/client-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -402,6 +402,9 @@ else()
     add_definitions (${Boost_LIB_DIAGNOSTIC_DEFINITIONS} -DBOOST_LOG_DYN_LINK)
 endif()
 
+option(KAA_RUNTIME_KEY_GENERATION "ON or OFF key pair generation" ON)
```
Comment (forGGe) [inline client/client-multi/client-cpp/CMakeLists.txt]: I think this description is too small to understand a feature

File: client/client-multi/client-cpp/impl/security/KeyUtils.cpp [hunk 2]
```diff
@@ -24,12 +24,14 @@ namespace kaa {
...
 KeyPair KeyUtils::generateKeyPair(std::size_t length)
 {
+#ifdef KAA_RUNTIME_KEY_GENERATION
     Botan::RSA_PrivateKey key(rng_, length);
     auto pemEncodedKey = Botan::PKCS8::PEM_encode(key);
     auto berEncodedKey = Botan::X509::BER_encode(key);
     return KeyPair(
             PublicKey(berEncodedKey.begin(),berEncodedKey.end()),
             PrivateKey(pemEncodedKey.begin(), pemEncodedKey.end()));
+#endif
```
Comment (forGGe) [inline client/client-multi/client-cpp/impl/security/KeyUtils.cpp]: This function **must** return

**Comment by forGGe** [comment]
Sophisticated review must be performed. Do not merge yet

File: client/client-multi/client-cpp/CMakeLists.txt (hunk 1 above)
Comment (rasendubi) [inline client/client-multi/client-cpp/CMakeLists.txt]: `"Enable RSA key generation at runtime"`

**Comment by Dubland** [comment]
Was fixed. Please review

File: client/client-multi/client-cpp/impl/KaaClient.cpp [hunk 3]
```diff
@@ -297,8 +297,12 @@ void KaaClient::initClientKeys()
     }
 
     if (regenerate) {
+#ifdef KAA_RUNTIME_KEY_GENERATION
         clientKeys_.reset(new KeyPair(utils.generateKeyPair(2048)));
         utils.saveKeyPair(*clientKeys_, publicKeyLocation, privateKeyLocation);
+#else
+        throw KaaException("Fauled to init Client Keys.");
```
Comment (ohamov) [inline client/client-multi/client-cpp/impl/KaaClient.cpp]: `s/Fauled/Failed/`, `s/init/initialize/`

File: client/client-multi/client-cpp/impl/KaaClient.cpp [hunk 4]
```diff
@@ -297,8 +297,13 @@ void KaaClient::initClientKeys()
     }
 
     if (regenerate) {
+#ifdef KAA_RUNTIME_KEY_GENERATION
         clientKeys_.reset(new KeyPair(utils.generateKeyPair(2048)));
         utils.saveKeyPair(*clientKeys_, publicKeyLocation, privateKeyLocation);
+#else
+        KAA_LOG_ERROR("KAA_RUNTIME_KEY_GENERATION is disabled. Generate keys and put them to the working directory.");
+        throw KaaException("Keus are missing.");
```
Comment (forGGe) [inline client/client-multi/client-cpp/impl/KaaClient.cpp]: `Keus` => `Keys`

**Comment by RostakaGmfun** [comment]
LGTM

**Comment by RostakaGmfun** [comment]
retest this please

**Comment by forGGe** [comment]
+

**Pull request merged**